# Metagenomics

1. [General introduction](#General-introduction)
2. [Schedule](#Schedule)
    - [Monday](#Monday)
    - [Tuesday](#Tuesday)
    - [Wednesday](#Wednesday)
    - [Thursday](#Thursday)
    - [Friday](#Friday)
3. [Target group](#target-group)
4. [Learning outcomes](#learning-outcomes)
5. [Installations](#installations)
6. [Internet connection](#internet-connection)
7. [Organizers and teachers](#Organizers-and-teachers)


## General introduction
Microbial community structure and diversity, and population structure are fundamental aspects to understand evolution, niche adaptation and demographic history of bacterial species. As NGS sequencing has become cost effective and accessible, sequencing populations of bacteria across the whole genome provides unprecedented resolution to investigate within-host evolution, transmission history, and population structure. Moreover, analysis of genetic content of microbial communities through metagenomics has become the mainstream methodology. During the one week course the students will learn and apply bioinformatic techniques to perform population genetics and study microbial communities with metagenomic approaches. The goal is to become familiar with the bioinformatic analysis tools and to be able to utilize them in own research after the course.

**Everything is possible thanks to the support of [CSC](http://www.csc.fi)**

## Schedule
The course will be held in Viikki at the Faculty of Agriculture and Forestry, University of Helsinki.  
Place: __Info Centre Korona, room Info4, 1st Floor__


### Monday  
__20.5.2019 9-17__  

| Time | Description |
| --- | --- |
| __*Morning*__ | [Introduction](https://drive.google.com/file/d/1Fr63zkAQ8EiG7J37taBD0GMLRHaQzBu-/view?usp=sharing) (__Jenni__) |
| __*Morning*__ | [Basic usage of computing services at CSC](https://drive.google.com/open?id=13Afjk2lDxx4kh2Jdd_5jIGrNcPqflIGJ) (__Antti__) |
| __*Morning*__ | [Installations for the course](MetagenomeInstallations.md) (__Antti__) |
| __*Morning*__ |  [Quality control and trimming of data](Day1/README.md#QC-and-trimming) hands-on, [Slides](https://drive.google.com/file/d/1qA8xxK2nwSqZo2lV3BohKQA_RlHW7QBb/view?usp=sharing) (__Jenni__)|
|  |    **Lunch** |
| __*Afternoon*__ | [Assembly](Day1/README.md#Assembly) hands-on, [Slides](https://drive.google.com/file/d/1EHsy1spesv3gIAALWWX7_1NBu9dwA37J/view?usp=sharing)(__Jenni__) |
| __*Afternoon*__ | [Read based annotation](https://github.com/karkman/MetagenomeCourse2019/tree/master/Day1#humann2) (__Katariina__) |


### Tuesday
__21.5.2019 8-17__  

| Time | Description |
| --- | --- |
| __*Morning*__ | [Assembly statistics](Day2/README.md#assembly-quality-statistics) (__Jenni/Antti__)|
| __*Morning*__ | [Getting files for Anvio](Day2/README.md#anvio) (__Tom, Antti, Jenni__)|
|  |    **Lunch** |
| __*Afternoon*__ | [Introduction to Anvio](https://www.dropbox.com/s/yh1yf9vfuxbi9gl/2019_Helsinki_Tom_Delmont_Teaching_Material.pdf?dl=0) (__Tom__)
| __*Afternoon*__ | [Anvio hands-on](Day3/README.md#02--describing-the-interface) (__Tom__)

### Wednesday
__22.5.2019 8-17__  

| Time | Description |
| --- | --- |
| __*Morning*__  | [Manual binning of MAGs with Anvio](Day3/README.md#02--describing-the-interface) (__Tom__)|
|  |    **Lunch** |
| __*Afternoon*__  |  |
| __*Afternoon*__  | [Taxonomic annotation of MAGs](Day3/README.md#11--what-do-we-do-with-these-genomes) (__Tom__) |

### Thursday
__23.5.2019 8-16.30__  

| Time | Description |
| --- | --- |
| __*Morning*__  | *MetaPhlan lecture* (__Tommi__)|
| __*Morning*__  | [Metaphlan profiles](Day4#metaphlan2) (__Tommi__)|
|  |    **Lunch** |
| __*Afternoon*__  | *HUMAnN2 lecture* (__Tommi__)|
| __*Afternoon*__  | [Humann2 hands on](Day5#humann2) (__Tommi__)|

### Friday
__24.5.2019 9-17__  

| Time | Description |
| --- | --- |
| 09:00-10:00 | **Tommi Vatanen**: The gut microbiome structure and function in early life - lessons from cohort studies NB. Infocenter Korona, lecture hall 3)|
| 10:30-12:00 | [Humann2 hands on cont'd](Day5#humann2) [and R for Humann2 results](R_for_Humann2) (__Tommi__)|
|  |    **Lunch** |
| __*Afternoon*__  | [Crash course in statistics of metagenomic data: Multivariate analysis and generalized linear models of overdispersed count data](R_for_overdispersed_data/) hands on, [Slides](https://drive.google.com/open?id=1FMBSI1df1fDFoDXud3BDMb7Z6EIST0g0) (__Katariina__)|
| __*Afternoon*__ | Wrapping up (__all__) |

## Target group
This course is targeted at PhD and MSc students who are interested in performing bacterial population analysis using whole-genome sequencing and/or using metagenomics for investigating microbial communities.

## Learning outcomes
QA/QC of the raw sequence data, cleaning and assembly; Gene-by-gene annotation; Visualization of data; Taxonomic classification; Functional assignment, Analyses of microbial community composition; Comparative metagenomics; MAG (MetagenomeAssembledGenome) analysis; Statistical methods for metagenome data analysis.

## Installations
[Installation instructions](MetagenomeInstallations.md)

## Internet connection
Inside the class room you can find both __eduroam__ and Univ. Helsinki __HUPnet__. Instructions on how to connect in [HelpDesk website.](https://helpdesk.it.helsinki.fi/en/instructions/logging-and-connections/networks/wireless-connections-university)

## Organizers and Teachers

### Organizers
* Docent Jenni Hultman, University of Helsinki, Finland
* MSc Katariina Pärnänen, University of Helsinki, Finland
* PhD Antti Karkman, University of Helsinki, Finland

### Visiting teachers
* Dr Tom Delmont, Genoscope, France
* Dr Tommi Vatanen, The University of Auckland, New Zealand
